src/nvim/eval/encode.c

```diff
@@ -288,6 +288,11 @@ int encode_read_from_list(ListReaderState *const state, char *const buf,
           : OK);
 }
 
+#ifdef __MINGW32__
+# undef fpclassify
+# define fpclassify __fpclassify
+#endif
```

Comment:
Can this go in src/nvim/os/win_defs.h ? We should try to avoid sprinkling these types of fixups across the codebase.

Reply:
In fact we have this in strings.c also. 

For now, I just moved this to the top of this file.